ext/hash/hash.stub.php

```diff
@@ -2,15 +2,15 @@
 
 /** @generate-function-entries */
 
-function hash(string $algo, string $data, bool $binary = false): string|false {}
+function hash(string $algo, string $data, bool $binary = false, array $args = null): string|false {}
```

Comment:
I think the `gen_stub.php` script should throw an exception because it requires the type to be nullable when the default value is `null`.

Also, I'm wondering why just not to make `[]` the default?

Reply:
It makes it simpler to read internally as it'd be copied into the context object. Using `h!` right now, so it would set the struct prop. The suggestion with `[]` works, too. In that case, would have to check for the elements number internally. Just passing an empty array doesn't seem expedient - if some array is passed, it should actually contain something. And because the parameter is optional and in most cases won't be used, there doesn't seem to be a need.

Otherwise, i haven't seen any errors about this. 

 Thanks.

Reply:
AFAIR, `<type> $foo = null` implies "nullability", so this LGTM.

Reply:
We try to avoid nullable parameters if there is a well-defined non-null default. Using `[]` would be preferable.

Reply:
Choosing `null` was just for the internal reasons, as explained above. I'll rework this, to still carry `NULL` internally if an empty args was passed. That's just interfacing the user land.

thanks.

Reply:
Fixed, using '[]' by default.

thanks.